src/integrations/prefect-redis/prefect_redis/filesystem.py

```diff
@@ -0,0 +1,157 @@
+from contextlib import asynccontextmanager
+from typing import AsyncGenerator, Optional, Union
+
+import redis.asyncio as redis
+from pydantic import Field
+from pydantic.types import SecretStr
+from typing_extensions import Self
+
+from prefect.filesystems import WritableFileSystem
+
+
+class RedisFilesystem(WritableFileSystem):
```

Comment:
I know this is copying what currently exists, but I think we should change the name since `RedisFilesystem` seems paradoxical. Doesn't need to be part of this PR.

Reply:
The original name was actually 'RedisCredentials'. This is the renamed version, `RedisFilesystem` is what the [issue](https://github.com/PrefectHQ/prefect/issues/12249) suggested. 

In what way do you think it's paradoxical? The point of the block is to treat redis as a filesystem, much the same way GCS or S3 work, so the name makes sense to me, but happy to consider alternatives.

Reply:
The tests are failing for this because it doesn't have redis running in CI. But I'm going to go ahead and merge this, and have a follow-up to get those tests running. We can talk about naming, etc. in slack.

Reply:
I thought that Redis isn't an object storage solution like S3 and GCS that uses file-like semantics, so it feels weird to call it a filesystem. I guess we've backed ourselves into a corner with the `Filesystem` interface name. My vote would be to name it `RedisDatabase` or something like that since other classes that implement a `Filesystem` interface are named things like `S3Bucket` and `AzureBlobStorageContainer` to try and make it clear how they map onto external concepts.